pkg/chunk/purger/table_provisioning.go

```diff
@@ -0,0 +1,30 @@
+package purger
+
+import (
+	"flag"
+
+	"github.com/cortexproject/cortex/pkg/chunk"
+)
+
+// TableProvisioningConfig holds config for table throuput and autoscaling. Currently only used by DynamoDB.
+type TableProvisioningConfig struct {
+	chunk.ActiveTableProvisionConfig `yaml:",inline"`
+	TableTags                        chunk.Tags `yaml:"tags"`
+}
+
+// RegisterFlags adds the flags required to config this to the given FlagSet.
+// Adding a separate RegisterFlags here instead of using it from embedded chunk.ActiveTableProvisionConfig to be able to manage defaults separately.
+// Defaults for WriteScale and ReadScale are shared for now to avoid adding further complexity since autoscaling is disabled anyways by default.
+func (cfg *TableProvisioningConfig) RegisterFlags(argPrefix string, f *flag.FlagSet) {
+	f.Int64Var(&cfg.ProvisionedWriteThroughput, argPrefix+".write-throughput", 1, "Table default write throughput. Supported by DynamoDB")
+	f.Int64Var(&cfg.ProvisionedReadThroughput, argPrefix+".read-throughput", 300, "Table default read throughput. Supported by DynamoDB")
+	f.BoolVar(&cfg.ProvisionedThroughputOnDemandMode, argPrefix+".enable-ondemand-throughput-mode", false, "Enables on demand throughput provisioning for the storage provider (if supported). Applies only to tables which are not autoscaled. Supported by DynamoDB")
+	f.Var(&cfg.TableTags, argPrefix+".tags", "Tag (of the form key=value) to be added to the tables. Supported by DynamoDB")
+
+	cfg.WriteScale.RegisterFlags(argPrefix+".write-throughput.scale", f)
+	cfg.ReadScale.RegisterFlags(argPrefix+".read-throughput.scale", f)
```

Comment:
```suggestion
	cfg.ActiveTableProvisionConfig.RegisterFlags(argPrefix, f)
	f.Var(&cfg.TableTags, argPrefix+".tags", "Tag (of the form key=value) to be added to the tables. Supported by DynamoDB")
```

Reply:
@pstibrany thanks for the review. The defaults in `ActiveTableProvisionConfig` are way too high considering the amount of traffic expected for tables storing delete request, which is why I am doing it separately here with lower defaults. Does it make sense?

Reply:
In that case, I would suggest that `cfg.ActiveTableProvisionConfig.RegisterFlags` takes defaults from `cfg.ActiveTableProvisionConfig`, and we set correct defaults in this method, and other place where `RegisterFlags` is called. I'd prefer if we can avoid duplicating all config option names and help strings. WDYT?